src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+public class Rectangle extends Figure {
+    private final double width;
+    private final double length;
+
+    public Rectangle(double width, double length) {
+        this.width = width;
+        this.length = length;
+        setColor(new ColorSupplier().getRandomColor());
```

Comment:
You can pass random values to the constructor of a figure instead of generating them inside figure classes. It's better to create an instance of `ColorSupplier` in `FigureSupplier` and pass the color as a parameter to the `Rectangle` constructor.